### ISSUE TITLE
Investigate SecurityHub policy that results in AccessDenied for config:GetComplianceDetailsByConfigRule

### ISSUE THREAD

**Issue by jakemulley** [ministryofjustice/modernisation-platform#68]
[AWS SecurityHub](https://aws.amazon.com/security-hub/) is enabled via the [modernisation-platform-terraform-baselines](https://github.com/ministryofjustice/modernisation-platform-terraform-baselines/blob/main/modules/securityhub/main.tf) Terraform module. It uses the [`AWSServiceRoleForSecurityHub` service-linked role](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-settingup.html). It integrates with [AWS Config](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-awsconfigrules.html) to configure a standard set of rules.

When SecurityHub assumes the service-linked role, and tries to perform `config:GetComplianceDetailsByConfigRule` on its own standard set of rules, the resource throws an AccessDenied error. This only happens on the SecurityHub-created rules, not any that are configured manually.

The error it throws is:
`User: arn:aws:sts::${accountId}:assumed-role/AWSServiceRoleForSecurityHub/securityhub is not authorized to perform: config:GetComplianceDetailsByConfigRule on resource: securityhub-${rule-name}`.

This full error stack is viewable in CloudWatch Logs if you search for `AccessDenied`.

Time box to 1 day.

**Comment by mikedanielcpt** [comment]
I have a similar issue, started in last week or so....out of nowhere. Is there a fix on this?

**Comment by bellbrothers** [comment]
I ran into the same error/symptom; the solution was to enable Config in the affected region.  The region was specified in the CloudWatch Log error. 

Config - like GuardDuty and SecurityHub - is Region based.   

Note - I came to this error via CloudFormation not the Terraform module.  In multiple accounts I turned on GuardDuty/SecurityHub but didn't turn on Config and got this error.  Errors disappeared after turning on Config.

I found this issue when looking for a solution and assume others will as well.
